### PR TITLE
Fix iOS - can't show Popup from a modal page

### LINKED ISSUE #1189: [Bug] iOS - can't show Popup from a modal page
### Description

On iOS, Navigation.ShowPopup() fails when called from a modally presented Page.  The same scenario works fine on Android.

### Steps to Reproduce

1. Call Navigation.PushModalAsync(new ModalPage()) to show ModalPage
2. On ModalPage, call Navigation.ShowPopup(new CustomPopup()) so show CustomPopup.

### Expected Behavior
Popup should display on iOS as it does on Android

### Actual Behavior
Popup failed to display with error "2021-04-10 12:04:13.936120+0100 XF_PopupModalIssue.iOS[5438:145915] [Presentation] Attempt to present <Xamarin_CommunityToolkit_UI_Views_PopupRenderer: 0x7fcd44e60000> on <Xamarin_Forms_Platform_iOS_PlatformRenderer: 0x7fcd44c25ba0> (from <Xamarin_Forms_Platform_iOS_PageRenderer: 0x7fcd4711aba0>) which is already presenting <Xamarin_Forms_Platform_iOS_ModalWrapper: 0x7fcd44d42e30>."

### Basic Information

- Version with issue: v1.1.0
- Last known good version: N/A
- IDE:
- Platform Target Frameworks: 
  - iOS:  14.4
  - Android: v9.0 (API Level 28)
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

### Workaround
None discovered

### Reproduction imagery

Screenshot 2021-04-10 at 12 09 34

### Reproduction Link
Attached zipped XF solution reproduces the issue
[XF_PopupModalIssue.zip](https://github.com/xamarin/XamarinCommunityToolkit/files/6289698/XF_PopupModalIssue.zip)

### PR DESCRIPTION
**Pull request by Neihlanda**
### Description of Change ###

Actually on iOS, the ViewController is the MainPage ViewController, with this changes if there is any Modal into the MainPage.Navigation.ModalStack, the ViewController is the last modal displayed ViewController.

### Bugs Fixed ###

- Fixes #1189

You can try it with the sample of the issue fixed here : [XF_PopupModalIssue_Fixed.zip](https://github.com/xamarin/XamarinCommunityToolkit/files/6657933/XF_PopupModalIssue_Fixed.zip) 

### API Changes ###

None

### Behavioral Changes ###

None

### PR Checklist ###

- [x] Has a linked Issue, and the Issue has been `approved`
- [x] Has tests (if omitted, state reason in description)
- [ ] Has samples (if omitted, state reason in description)
- [ ] Rebased on top of main at time of PR
- [ ] Changes adhere to coding standard

- [ ] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

### CODE REVIEW

**Comment by pictos** [comment]
@Neihlanda thanks for this PR. Can you add a test page to confirm that this issue doesn't exist anymore? You can set up the test [page here](https://github.com/xamarin/XamarinCommunityToolkit/tree/main/samples/XCT.Sample/Pages/TestCases)

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/iOS/PopupRenderer.ios.cs [hunk 1]
```diff
@@ -133,7 +133,16 @@ void CreateControl()
...
+			IVisualElementRenderer currentPageRenderer;
+			if (Application.Current.MainPage?.Navigation?.ModalStack.Count > 0)
+			{
+				var index = Application.Current.MainPage.Navigation.ModalStack.Count - 1;
+				currentPageRenderer = Platform.GetRenderer(Application.Current.MainPage.Navigation.ModalStack[index]);
+			}
+			else
+			{
+				currentPageRenderer = Platform.GetRenderer(Application.Current.MainPage);
+			}
```
Comment (pictos) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/iOS/PopupRenderer.ios.cs:145]:
```suggestion
			var modalStackCount = Application.Current.MainPage?.Navigation?.ModalStack?.Count ?? 0;
			var mainPage = Application.Current.MainPage;
			if (modalStackCount > 0)
			{
				var index = modalStackCount - 1;
				currentPageRenderer = Platform.GetRenderer(mainPage!.Navigation!.ModalStack![index]);
			}
			else
			{
				currentPageRenderer = Platform.GetRenderer(mainPage);
			}
```

**Comment by Neihlanda** [comment]
@pictos  Thanks for the review, I've set a test case as u asking. The tests are in "Popups" folder like it's done in the "Views" folders

**Comment by jfversluis** [comment]
Wonderful @Neihlanda thank you!

@ahoefling any thoughts?

**Review by jfversluis — APPROVED** [review]

**Pull request merged**